core/domain/platform_parameter_domain_test.py

```diff
@@ -456,6 +456,50 @@ def test_evaluate_lte_version_filter_with_large_version_returns_false(self):
         self.assertFalse(filter_domain.evaluate(
             self._create_example_context(app_version='2.0.0')))
 
+    def test_evaluate_version_flavor_filter_with_matched_flavor_returns_true(
+            self):
+        filter_domain = (
+            parameter_domain
+            .PlatformParameterFilter.from_dict(
+                {'type': 'app_version_flavor', 'conditions': [('=', 'beta')]}
+            )
+        )
+        self.assertTrue(filter_domain.evaluate(
+            self._create_example_context(app_version='1.10.0-abcdef-beta')))
+
+    def test_evaluate_version_flavor_filter_with_unmatched_flavor_returns_false(
```

Comment:
Please also add tests for comparison operators & for each flavor that we want to ensure we support.

Reply:
Done, added tests for all supported flavors. -- for comparison operators, as flavor filters are independent of version comparison filters, I think we don't need to test these two together.

Reply:
Oops, I meant to leave a comment suggesting we add conditional support. In particular, the inheritance flow would be:

test < alpha < beta < release
unknown != {test, alpha, beta, release} (e.g. unknown always fails the filter)

Please make sure there are tests for these, too.

Reply:
Done, added comparison support to the flavor filter.